src/CoreWCF.UnixDomainSocket/src/CoreWCF/Channels/UnixDomainSocketListenOptions.cs

```diff
@@ -0,0 +1,129 @@
+        // add to this class. If there are multiple endpoints using named pipes, on .NET Framework the first endpoint would establish the settings
+        // and subsequent ones would need validate that the settings matched. By moving the settings needed to be considered for a shared listener
+        // to NamedPipeListenOptions, we remove the need to reconcile multiple bindings and just set it on the classes which are responsible
+        // for that part of the IO.
+        private readonly List<Func<ConnectionDelegate, ConnectionDelegate>> _middleware = new List<Func<ConnectionDelegate, ConnectionDelegate>>();
+        private HostNameComparisonMode _hostNameComparisonMode;
+        private int _connectionBufferSize;
+        private int _maxPendingAccepts;
+        private List<SecurityIdentifier> _allowedUsers;
+
+        internal UnixDomainSocketListenOptions(Uri baseAddress)
+        {
+            UnixDomainSocketUri.Validate(baseAddress);
+            BaseAddress = baseAddress;
+            FilePath = baseAddress.AbsolutePath;
+            _connectionBufferSize = ConnectionOrientedTransportDefaults.ConnectionBufferSize;
+            _hostNameComparisonMode = ConnectionOrientedTransportDefaults.HostNameComparisonMode;
+            _maxPendingAccepts = ConnectionOrientedTransportDefaults.GetMaxPendingAccepts();
+        }
+
+        public List<SecurityIdentifier> AllowedUsers
+        {
+            get
+            {
+                if (_allowedUsers == null)
+                {
+                    _allowedUsers = new List<SecurityIdentifier>();
+                }
+
+                return _allowedUsers;
+            }
+        }
+
+        // This property will return null if the AllowedUsers getter hasn't been used
+        internal List<SecurityIdentifier> InternalAllowedUsers => _allowedUsers;
+
+        public IServiceProvider ApplicationServices => UnixDomainSocketServerOptions?.ApplicationServices;
+        public Uri BaseAddress { get; }
+        public string FilePath { get; }
+
+        public int ConnectionBufferSize
+        {
+            get
+            {
+                return _connectionBufferSize;
+            }
+            set
+            {
+                if (value < 0)
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentOutOfRangeException(nameof(value), value, SRCommon.ValueMustBeNonNegative));
+                }
+
+                _connectionBufferSize = value;
+            }
+        }
+
+        public HostNameComparisonMode HostNameComparisonMode
+        {
+            get
+            {
+                return _hostNameComparisonMode;
+            }
+
+            set
+            {
+                HostNameComparisonModeHelper.Validate(value);
+                _hostNameComparisonMode = value;
+            }
+        }
+
+        public int MaxPendingAccepts
+        {
```

Comment:
I don't think these values are currently being propagated to the Kestrel listener.

Reply:
There is no use... I am removing.. let me know if you think otherwise.